pkgs/applications/misc/ticker/default.nix

```diff
@@ -1,18 +1,28 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
 
 buildGoModule rec {
   pname = "ticker";
   version = "3.1.7";
 
   src = fetchFromGitHub {
     owner = "achannarasappa";
-    repo = "ticker";
+    repo = pname;
     rev = "v${version}";
     sha256 = "sha256-OA01GYp6E0zsEwkUUtvpmvl0y/YTXChl0pwIKozB4Qg=";
   };
 
   vendorSha256 = "sha256-aUBj7ZGWBeWc71y1CWm/KCw+El5TwH29S+KxyZGH1Zo=";
 
+  buildFlagsArray = [
```

Comment:
There is actually a nixpkgs-hammering check that suggest to set this array with bash in preBuild due to buggy conversion from nix lists to bash arrays.

Reply:
nixpkgs-hammering explanation: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md

---

In my experience trying to set `-ldflags="-s -w -X something=yo"` that way is much much more of a pain.

Do you have an example of this setup correctly?

Reply:
https://github.com/NixOS/nixpkgs/blob/66ba4dc61fb073eaf9ddc89d0eb212264dfb29ed/pkgs/tools/misc/noti/default.nix#L20

Reply:
Thanks

Reply:
ok yeah `buildFlagsArray+=("-ldflags" "-s -w -X whatever=v${version}")` worked fine

Reply:
It would be good if the [documentation](https://nixos.org/manual/nixpkgs/unstable/#ssec-go-common-attributes) is updated as there is mentioned:

```bash
  buildFlagsArray = [
    "-ldflags=-X main.Version=${version} -X main.Commit=${version}"
  ];
```
and 

```bash
  buildFlagsArray = ''
    -ldflags=
    -X main.Version=${version}
    -X main.Commit=${version}
  '';
```
as examples. This one is also  broadly used:

```bash
  buildFlagsArray = [
    "-ldflags="
    "-s"
    "-w"
    "-X main.Version=${version}"
  ];
```